pkg/snapshot/snapshot.go

```diff
@@ -155,7 +155,16 @@ func (s *Snapshotter) scanFullFilesystem() ([]string, []string, error) {
 	// for example the hashing function that determines if files are equal uses the mtime of the files,
 	// which can lag if sync is not called. Unfortunately there can still be lag if too much data needs
 	// to be flushed or the disk does its own caching/buffering.
-	syscall.Sync()
+	dir, err := os.Open(s.directory)
+	if err != nil {
+		return nil, nil, err
+	}
+	defer dir.Close()
+	// 306 is the number for sys_syncfs
+	_, _, errno := syscall.Syscall(306, dir.Fd(), 0, 0)
```

Comment:
nit: Can you add some OS related checks to ensure that this is only done for operating systems that support sys_syncfs.  Perhaps look at `runtime.GOOS` and only do the sys_syncfs logic for linux OS?  Admittedly kaniko is mostly run in a container on top of linux containerization but just for completeness/future-proofing this check might make sense

Reply:
Sounds good. Thanks for the suggestion.

Reply:
Ah, `306` is also amd64 specific. I think I can leverage golang.org/x/sys/unix.